animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ModelImpl.java

```diff
@@ -144,6 +147,23 @@ public ModelImpl(EntityManagerFactory emf) {
                                                           updatedBy);
     }
 
+    /* (non-Javadoc)
+     * @see com.chiralbehaviors.CoRE.meta.Model#executeAs(com.chiralbehaviors.CoRE.security.AuthenticatedPrincipal, java.util.concurrent.Callable)
+     */
+    @Override
+    public <V> V executeAs(AuthenticatedPrincipal principal,
+                           Callable<V> function) throws Exception {
+        V value = null;
+        AuthenticatedPrincipal previous = currentPrincipal.get();
```

Comment:
How deep can the stack go? It looks like you can only pop two contexts (current and previous) from a particular modelimpl, which is tied to an EntityManager. How careful will we have to be in managing Models?